File: android-app/app/src/main/java/com/example/mobileapp/models/Purchase.kt
package com.example.mobileapp.models

data class Purchase(
    val id: String,
    val purchaseId: String,
    val userId: String,
    val productId: Int,
    val quantity: Int,
    val unitPrice: Int,
    val totalAmount: Int,
    val purchasedAt: String
)

data class PurchaseRequest(
    val productId: Int,
    val quantity: Int
)

data class PurchaseResponse(
    val data: Purchase,
    val timestamp: String
)

File: src/ios-app/MobileApp/MobileApp/Models/Purchase.swift
//
//  Purchase.swift
//  MobileApp
//
//  Created on 2025-01-08.
//

import Foundation

struct Purchase: Codable, Identifiable {
    let id: String
    let purchaseId: String
    let userId: String
    let productId: Int
    let quantity: Int
    let unitPrice: Int
    let totalAmount: Int
    let purchasedAt: String
}

struct PurchaseRequest: Codable {
    let productId: Int
    let quantity: Int
}

struct PurchaseResponse: Codable {
    let data: Purchase
    let timestamp: String
}
